src/main/java/com/forgeessentials/chat/ModuleChat.java

```diff
@@ -46,202 +66,446 @@
 public class ModuleChat
 {
 
+    public static final String CONFIG_FILE = "Chat";
+
     public static final String CONFIG_CATEGORY = "Chat";
 
-    public static boolean welcomeNewPlayers;
-    public static String welcomeNewPlayerMsg;
+    // @formatter:off
+    static final Pattern URL_PATTERN = Pattern.compile(
+            //         schema                          ipv4            OR           namespace                 port     path         ends
+            //   |-----------------|        |-------------------------|  |----------------------------|    |---------| |--|   |---------------|
+            "((?:(?:http|https):\\/\\/)?(?:(?:[0-9]{1,3}\\.){3}[0-9]{1,3}|(?:[-\\w_\\.]{1,}\\.[a-z]{2,}?))(?::[0-9]{1,5})?.*?(?=[!\"\u00A7 \n]|$))",
+            Pattern.CASE_INSENSITIVE);
+    // @formatter:on
 
-    @FEModule.ModuleDir
-    public static File moduleDir;
+    public static interface ChatReplacer
+    {
+        public Object getReplacement(EntityPlayerMP player);
+    }
 
-    public static PrintWriter chatLog;
+    public static final Map<String, ChatReplacer> chatReplacements = new HashMap<>();
 
-    public static PrintWriter cmdLog;
+    public static final Map<String, String> chatConstReplacements = new HashMap<>();
 
-    public static File logdir;
+    @FEModule.Instance
+    public static ModuleChat instance;
 
-    public static boolean connectToIRC;
+    private PrintWriter logWriter;
 
-    private MailSystem mailsystem;
+    public Censor censor;
 
-    @SuppressWarnings("unused")
-    private PlayerEventHandler ircPlayerHandler;
+    public IrcHandler ircHandler;
 
-    private AutoMessage autoMessage;
+    public TimedMessageHandler timedMessageHandler;
 
     @SubscribeEvent
-    public void load(FEModuleInitEvent e)
+    public void moduleLoad(FEModuleInitEvent e)
     {
-        // Register configuration
-        ForgeEssentials.getConfigManager().registerLoader(CONFIG_CATEGORY, new ConfigChat());
+        MinecraftForge.EVENT_BUS.register(this);
+        FMLCommonHandler.instance().bus().register(this);
 
-        MinecraftForge.EVENT_BUS.register(new ChatFormatter());
-        MinecraftForge.EVENT_BUS.register(new CommandMuter());
+        ForgeEssentials.getConfigManager().registerLoader(CONFIG_FILE, new ChatConfig());
 
-        if (connectToIRC)
-        {
-            ircPlayerHandler = new PlayerEventHandler();
-            MinecraftForge.EVENT_BUS.register(new IRCChatFormatter());
-        }
+        timedMessageHandler = new TimedMessageHandler();
+        ircHandler = new IrcHandler();
+        censor = new Censor();
+
+        setupChatReplacements();
     }
 
-    @SubscribeEvent
-    public void postLoad(FEModulePostInitEvent e)
+    public void setupChatReplacements()
     {
-        mailsystem = new MailSystem();
-
+        chatConstReplacements.put("smile", "\u263A");
```

Comment:
I moved it from FunctionHelper over here.
If you want to change it, do it yourself once the PR is merged.
I don't have time for these small and pretty useless changes.